internal_wasm/utils/utils.go

```diff
@@ -9,25 +9,27 @@ import (
 // See https://stackoverflow.com/questions/68426700/how-to-wait-a-js-async-function-from-golang-wasm
 func Await(awaitable js.Value) ([]js.Value, []js.Value) {
 	then := make(chan []js.Value)
-	defer close(then)
 	thenFunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
 		then <- args
 		return nil
 	})
+	// defers are called LIFO!
+	// This will `close` before `Release()`
+	defer thenFunc.Release()
```

Comment:
Unknown, but how does the `.Call("then" …).Call("catch")` work? Do we need to wait to release `thenFunc` until `catch` has been called? Or no?

Reply:
All `defers` will be called after `return`, so they clean themselves up automatically.